newlib/libc/machine/riscv/strcmp.S

```diff
@@ -37,21 +37,24 @@ strcmp:
   and   a4, a4, SZREG-1
   bnez  a4, .Lmisaligned
 
-#if SZREG == 4
-  li a5, 0x7f7f7f7f
-#else
-  ld a5, mask
-#endif
-
   .macro check_one_word i n
     REG_L a2, \i*SZREG(a0)
     REG_L a3, \i*SZREG(a1)
+#ifdef __riscv_zbb
+    orc.b t0, a2
+#else
+
+#if SZREG == 4
```

Comment:
This moves the load of 'a5' inside the macro so that it will be replicated across all macro instances and get executed inside the loop. Please leave the old code path intact for devices without Zbb.

Reply:
I updated the commit, please review